### PR TITLE
Utilities: Port useradd and groupdel to LibMain

### PR DESCRIPTION
**Pull request by TheOddGarlic**
This PR ports useradd and groupdel to LibMain

### CODE REVIEW

File: Userland/Utilities/groupdel.cpp [hunk 1]
```diff
@@ -68,20 +60,22 @@ int main(int argc, char** argv)
...
     };
 
     ArmedScopeGuard unlink_temp_files_guard = [&] {
         unlink_temp_files();
     };
 
-    auto temp_group_fd = mkstemp(temp_group);
-    if (temp_group_fd == -1) {
+    auto mkstemp_or_error = Core::System::mkstemp(temp_group);
+    if (mkstemp_or_error.is_error()) {
```
Comment (linusg) [inline Userland/Utilities/groupdel.cpp:72]: The whole point of these `TRY()`-able wrappers is that we don't have to manually make an error message and return :)

This, and everything else of the same pattern, should just be `auto temp_group_fd = TRY(Core::System::mkstemp(temp_group));`

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Utilities/groupdel.cpp (hunk 1 above)
Comment (TheOddGarlic) [reply Userland/Utilities/groupdel.cpp:72]: I thought keeping a more verbose error message would be better (for instance this one has "failed to create temporary group file"). I will change these accordingly.

**Review by kennethmyhra — CHANGES_REQUESTED** [review]
The fixup commit: `groupdel: Handle errors more idiomatically with TRY` should be squashed into the previous commit

**Review by kennethmyhra — APPROVED** [review]

**Comment by kennethmyhra** [comment]
@TheOddGarlic this needs a rebase. And you should probably post it in #code-review in discord to get attention from a maintainer.

**Comment by TheOddGarlic** [comment]
Okay I'll do that 👍

**Pull request merged**